trd-digital/trd-news-interactive-maps
# [TRD-3609] Create concept NYC construction project map

## Summary
The branch contains the HTML and support files for the approved NYC construction jobs (i.e. projects) map. The map allow an user to use a slider to filter the construction jobs by the quarter during which the job was approved.

## Testing Instruction
1. Run a local web server
 ```bash
 npm run serve
 ```
2. Open the NYC approved construction map on a browser using the localhost URL [http://localhost:8080/new-york-city-approved-construction-projects/new-york-city-approved-construction-projects.html](http://localhost:8080/new-york-city-approved-construction-projects/new-york-city-approved-construction-projects.html). Alternately, one can replace `localhost` in the local host URL with the host address that appears in the `npm run serve` output.
   ![Screenshot 2024-09-10 at 12 49 11 PM](https://github.com/user-attachments/assets/033fdea5-8ea7-4c70-8632-206e8b022481)
For example, based on the output above, the alternate map URL will be http://192.168.29.11:8080/new-york-city-approved-construction-projects/new-york-city-approved-construction-projects.html.

3. Please wait at least 60 seconds. Confirm that the map, points, legend and slider appear after 60 seconds.
   <img width="1709" alt="Screenshot 2024-09-09 at 10 51 47 PM" src="https://github.com/user-attachments/assets/b19f4bc7-d668-43e3-9c24-e970c4ce945d">

4. Interact with the slider at the map's bottom center to confirm that the map filters the displayed point.
   4a. Go to the slider.
     <img width="362" alt="Screenshot 2024-09-11 at 11 04 08 AM" src="https://github.com/user-attachments/assets/4939295a-fc05-4f86-a2f5-65da16638e34">
   4b. Press and move the slider button to the right. Confirm that the displayed points and filing quarter text change (from `2023Q4` to `2024Q1`).
    <img width="537" alt="Screenshot 2024-09-11 at 11 14 49 AM" src="https://github.com/user-attachments/assets/3396e019-0849-476c-9c87-e2ef2f2432ac">
   4c. Press and move the button to the right again. Confirm that the displayed points change again and filing quarter text change from `2024Q1 to `2024Q2`.
    <img width="699" alt="Screenshot 2024-09-11 at 11 22 05 AM" src="https://github.com/user-attachments/assets/6ce0a88f-d93d-4c07-81db-30bf394e7aa5">
   4d. Press and move the button to the right again. Confirm that the displayed points change again and filing quarter text change from `2024Q2 to `2024Q3`.
    <img width="647" alt="Screenshot 2024-09-11 at 11 23 56 AM" src="https://github.com/user-attachments/assets/8b135a37-f97c-44cc-90dd-ec2cee325bc2">

     Below is a link to a video show how the slider should work when interacting with it.
 https://github.com/user-attachments/assets/e29b76d3-ee04-4254-b3d2-812a83ff02a8


5. Click on a point to see if the left info panel appears.
   <img width="579" alt="Screenshot 2024-09-10 at 1 01 40 PM" src="https://github.com/user-attachments/assets/5a994e83-0b86-4093-badd-c0f036c825c0">
